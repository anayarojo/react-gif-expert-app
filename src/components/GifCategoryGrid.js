import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifCategoryItem from './GifCategoryItem';

const GifCategoryGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>
            { loading && <p>Loading...</p> }
            <div className="card-grid">
            {
                images.map((img) => (
                    <GifCategoryItem
                        key={ img.id }
                        { ...img }
                    />
                ))
            }
            </div>
        </>
    )
}

GifCategoryGrid.propTypes = {
    category: PropTypes.string.isRequired
};

export default GifCategoryGrid;
