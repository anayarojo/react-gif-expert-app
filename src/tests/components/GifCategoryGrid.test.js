import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';

import GifCategoryGrid from '../../components/GifCategoryGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test for <GifCategoryGrid />', () => {
    
    const category = 'One Punch Man';

    test('Should show <GifCategoryGrid /> successfully', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifCategoryGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    });


    test('Sould show items when useFetchGifs() load images', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything/thing.jpg',
            title: 'Anything'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifCategoryGrid category={ category } />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifCategoryItem').length).toBe(gifs.length);
    });

});
