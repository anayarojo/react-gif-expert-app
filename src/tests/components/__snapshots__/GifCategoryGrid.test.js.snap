// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifCategoryGrid /> Should show <GifCategoryGrid /> successfully 1`] = `
<Fragment>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
